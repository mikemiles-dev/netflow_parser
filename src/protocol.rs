use nom_derive::*;
use serde::Serialize;

#[repr(u8)]
#[derive(Debug, PartialEq, Eq, Clone, Copy, PartialOrd, Ord, Serialize, Nom)]
pub enum ProtocolTypes {
    Hopopt = 0,
    Icmp = 1,
    Igmp = 2,
    Ggp = 3,
    Ipv4 = 4,
    St = 5,
    Tcp = 6,
    Cbt = 7,
    Egp = 8,
    Igp = 9,
    Bbcrccmon = 10,
    Nvpii = 11,
    Pup = 12,
    Argus = 13,
    Emcon = 14,
    Xnet = 15,
    Chaos = 16,
    Udp = 17,
    Mux = 18,
    Dcnmeas = 19,
    Hmp = 20,
    Prm = 21,
    Xnxidp = 22,
    Trunk1 = 23,
    Trunk2 = 24,
    Leaf1 = 25,
    Leaf2 = 26,
    Rdp = 27,
    Irtp = 28,
    Isotp4 = 29,
    Netblt = 30,
    Mfensp = 31,
    Meritinp = 32,
    Dccp = 33,
    Threepc = 34,
    Idpr = 35,
    Xtp = 36,
    Ddp = 37,
    Idprcmtp = 38,
    Tppp = 39,
    Il = 40,
    Ipv6 = 41,
    Sdrp = 42,
    Ipv6Route = 43,
    Ipv6Frag = 44,
    Idrp = 45,
    Rsvp = 46,
    Gre = 47,
    Dsr = 48,
    Bna = 49,
    Esp = 50,
    Ah = 51,
    Inlsp = 52,
    Swipe = 53,
    Narp = 54,
    Mobile = 55,
    Tlsp = 56,
    Skip = 57,
    Ipv6Icmp = 58,
    Ipv6Nonxt = 59,
    Ipv6Opts = 60,
    Anydistributedprotocol = 61,
    Cftp = 62,
    Anylocalnetwork = 63,
    Satexpak = 64,
    Kryptolan = 65,
    Rvd = 66,
    Ippc = 67,
    Anydistributedfilesystem = 68,
    Satmon = 69,
    Visa = 70,
    Ipcv = 71,
    Cpnx = 72,
    Cphb = 73,
    Wsn = 74,
    Pvp = 75,
    Brsatmon = 76,
    Sunnd = 77,
    Wbmon = 78,
    Wbexpak = 79,
    Isoip = 80,
    Vmtp = 81,
    Securevmtp = 82,
    Vines = 83,
    Iptm = 84,
    Nsfnetigp = 85,
    Dgp = 86,
    Tcf = 87,
    Eigrp = 88,
    Ospfigp = 89,
    Spriterpc = 90,
    Larp = 91,
    Mtp = 92,
    Ax25 = 93,
    Ipip = 94,
    Micp = 95,
    Sccsp = 96,
    Etherip = 97,
    Encap = 98,
    Anyprivateencryptionscheme = 99,
    Gmtp = 100,
    Ifmp = 101,
    Pnni = 102,
    Pim = 103,
    Aris = 104,
    Scps = 105,
    Qnx = 106,
    An = 107,
    Ipcomp = 108,
    Snp = 109,
    Compaqpeer = 110,
    Ipxinip = 111,
    Vrrp = 112,
    Pgm = 113,
    Any0Hopprotocol = 114,
    L2Tp = 115,
    Ddx = 116,
    Iatp = 117,
    Stp = 118,
    Srp = 119,
    Uti = 120,
    Smp = 121,
    Sm = 122,
    Ptp = 123,
    Isisoveripv4 = 124,
    Fire = 125,
    Crtp = 126,
    Crudp = 127,
    Sscopmce = 128,
    Iplt = 129,
    Sps = 130,
    Pipe = 131,
    Sctp = 132,
    Fc = 133,
    Rsvpe2Eignore = 134,
    Mobilityheader = 135,
    Udplite = 136,
    Mplsinip = 137,
    Manet = 138,
    Hip = 139,
    Shim6 = 140,
    Wesp = 141,
    Rohc = 142,
    Ethernet = 143,
    Aggfrag = 144,
    Unknown,
    Reserved = 255,
}

impl From<u8> for ProtocolTypes {
    fn from(item: u8) -> Self {
        match item {
            1 => ProtocolTypes::Hopopt,
            2 => ProtocolTypes::Igmp,
            3 => ProtocolTypes::Ggp,
            4 => ProtocolTypes::Ipv4,
            5 => ProtocolTypes::St,
            6 => ProtocolTypes::Tcp,
            7 => ProtocolTypes::Cbt,
            8 => ProtocolTypes::Egp,
            9 => ProtocolTypes::Igp,
            10 => ProtocolTypes::Bbcrccmon,
            11 => ProtocolTypes::Nvpii,
            12 => ProtocolTypes::Pup,
            13 => ProtocolTypes::Argus,
            14 => ProtocolTypes::Emcon,
            15 => ProtocolTypes::Xnet,
            16 => ProtocolTypes::Chaos,
            17 => ProtocolTypes::Udp,
            18 => ProtocolTypes::Mux,
            19 => ProtocolTypes::Dcnmeas,
            20 => ProtocolTypes::Hmp,
            21 => ProtocolTypes::Prm,
            22 => ProtocolTypes::Xnxidp,
            23 => ProtocolTypes::Trunk1,
            24 => ProtocolTypes::Trunk2,
            25 => ProtocolTypes::Leaf1,
            26 => ProtocolTypes::Leaf2,
            27 => ProtocolTypes::Rdp,
            28 => ProtocolTypes::Irtp,
            29 => ProtocolTypes::Isotp4,
            30 => ProtocolTypes::Netblt,
            31 => ProtocolTypes::Mfensp,
            32 => ProtocolTypes::Meritinp,
            33 => ProtocolTypes::Dccp,
            34 => ProtocolTypes::Threepc,
            35 => ProtocolTypes::Idpr,
            36 => ProtocolTypes::Xtp,
            37 => ProtocolTypes::Ddp,
            38 => ProtocolTypes::Idprcmtp,
            39 => ProtocolTypes::Tppp,
            40 => ProtocolTypes::Il,
            41 => ProtocolTypes::Ipv6,
            42 => ProtocolTypes::Sdrp,
            43 => ProtocolTypes::Ipv6Route,
            44 => ProtocolTypes::Ipv6Frag,
            45 => ProtocolTypes::Idrp,
            46 => ProtocolTypes::Rsvp,
            47 => ProtocolTypes::Gre,
            48 => ProtocolTypes::Dsr,
            49 => ProtocolTypes::Bna,
            50 => ProtocolTypes::Esp,
            51 => ProtocolTypes::Ah,
            52 => ProtocolTypes::Inlsp,
            53 => ProtocolTypes::Swipe,
            54 => ProtocolTypes::Narp,
            55 => ProtocolTypes::Mobile,
            56 => ProtocolTypes::Tlsp,
            57 => ProtocolTypes::Skip,
            58 => ProtocolTypes::Ipv6Icmp,
            59 => ProtocolTypes::Ipv6Nonxt,
            60 => ProtocolTypes::Ipv6Opts,
            61 => ProtocolTypes::Anydistributedprotocol,
            62 => ProtocolTypes::Cftp,
            63 => ProtocolTypes::Anylocalnetwork,
            64 => ProtocolTypes::Satexpak,
            65 => ProtocolTypes::Kryptolan,
            66 => ProtocolTypes::Rvd,
            67 => ProtocolTypes::Ippc,
            68 => ProtocolTypes::Anydistributedfilesystem,
            69 => ProtocolTypes::Satmon,
            70 => ProtocolTypes::Visa,
            71 => ProtocolTypes::Ipcv,
            72 => ProtocolTypes::Cpnx,
            73 => ProtocolTypes::Cphb,
            74 => ProtocolTypes::Wsn,
            75 => ProtocolTypes::Pvp,
            76 => ProtocolTypes::Brsatmon,
            77 => ProtocolTypes::Sunnd,
            78 => ProtocolTypes::Wbmon,
            79 => ProtocolTypes::Wbexpak,
            80 => ProtocolTypes::Isoip,
            81 => ProtocolTypes::Vmtp,
            82 => ProtocolTypes::Securevmtp,
            83 => ProtocolTypes::Vines,
            84 => ProtocolTypes::Iptm,
            85 => ProtocolTypes::Nsfnetigp,
            86 => ProtocolTypes::Dgp,
            87 => ProtocolTypes::Tcf,
            88 => ProtocolTypes::Eigrp,
            89 => ProtocolTypes::Ospfigp,
            90 => ProtocolTypes::Spriterpc,
            91 => ProtocolTypes::Larp,
            92 => ProtocolTypes::Mtp,
            93 => ProtocolTypes::Ax25,
            94 => ProtocolTypes::Ipip,
            95 => ProtocolTypes::Micp,
            96 => ProtocolTypes::Sccsp,
            97 => ProtocolTypes::Etherip,
            98 => ProtocolTypes::Encap,
            99 => ProtocolTypes::Anyprivateencryptionscheme,
            100 => ProtocolTypes::Gmtp,
            101 => ProtocolTypes::Ifmp,
            102 => ProtocolTypes::Pnni,
            103 => ProtocolTypes::Pim,
            104 => ProtocolTypes::Aris,
            105 => ProtocolTypes::Scps,
            106 => ProtocolTypes::Qnx,
            107 => ProtocolTypes::An,
            108 => ProtocolTypes::Ipcomp,
            109 => ProtocolTypes::Snp,
            110 => ProtocolTypes::Compaqpeer,
            111 => ProtocolTypes::Ipxinip,
            112 => ProtocolTypes::Vrrp,
            113 => ProtocolTypes::Pgm,
            114 => ProtocolTypes::Any0Hopprotocol,
            115 => ProtocolTypes::L2Tp,
            116 => ProtocolTypes::Ddx,
            117 => ProtocolTypes::Iatp,
            118 => ProtocolTypes::Stp,
            119 => ProtocolTypes::Srp,
            120 => ProtocolTypes::Uti,
            121 => ProtocolTypes::Smp,
            122 => ProtocolTypes::Sm,
            123 => ProtocolTypes::Ptp,
            124 => ProtocolTypes::Isisoveripv4,
            125 => ProtocolTypes::Fire,
            126 => ProtocolTypes::Crtp,
            127 => ProtocolTypes::Crudp,
            128 => ProtocolTypes::Sscopmce,
            129 => ProtocolTypes::Iplt,
            130 => ProtocolTypes::Sps,
            131 => ProtocolTypes::Pipe,
            132 => ProtocolTypes::Sctp,
            133 => ProtocolTypes::Fc,
            134 => ProtocolTypes::Rsvpe2Eignore,
            135 => ProtocolTypes::Mobilityheader,
            136 => ProtocolTypes::Udplite,
            137 => ProtocolTypes::Mplsinip,
            138 => ProtocolTypes::Manet,
            139 => ProtocolTypes::Hip,
            140 => ProtocolTypes::Shim6,
            141 => ProtocolTypes::Wesp,
            142 => ProtocolTypes::Rohc,
            143 => ProtocolTypes::Ethernet,
            144 => ProtocolTypes::Reserved,
            _ => ProtocolTypes::Unknown,
        }
    }
}
