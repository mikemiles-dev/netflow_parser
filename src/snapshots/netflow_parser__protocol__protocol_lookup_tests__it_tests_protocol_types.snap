---
source: src/protocol.rs
expression: protocols
---
- Unknown
- Hopopt
- Igmp
- Ggp
- Ipv4
- St
- Tcp
- Cbt
- Egp
- Igp
- Bbcrccmon
- Nvpii
- Pup
- Argus
- Emcon
- Xnet
- Chaos
- Udp
- Mux
- Dcnmeas
- Hmp
- Prm
- Xnxidp
- Trunk1
- Trunk2
- Leaf1
- Leaf2
- Rdp
- Irtp
- Isotp4
- Netblt
- Mfensp
- Meritinp
- Dccp
- Threepc
- Idpr
- Xtp
- Ddp
- Idprcmtp
- Tppp
- Il
- Ipv6
- Sdrp
- Ipv6Route
- Ipv6Frag
- Idrp
- Rsvp
- Gre
- Dsr
- Bna
- Esp
- Ah
- Inlsp
- Swipe
- Narp
- Mobile
- Tlsp
- Skip
- Ipv6Icmp
- Ipv6Nonxt
- Ipv6Opts
- Anydistributedprotocol
- Cftp
- Anylocalnetwork
- Satexpak
- Kryptolan
- Rvd
- Ippc
- Anydistributedfilesystem
- Satmon
- Visa
- Ipcv
- Cpnx
- Cphb
- Wsn
- Pvp
- Brsatmon
- Sunnd
- Wbmon
- Wbexpak
- Isoip
- Vmtp
- Securevmtp
- Vines
- Iptm
- Nsfnetigp
- Dgp
- Tcf
- Eigrp
- Ospfigp
- Spriterpc
- Larp
- Mtp
- Ax25
- Ipip
- Micp
- Sccsp
- Etherip
- Encap
- Anyprivateencryptionscheme
- Gmtp
- Ifmp
- Pnni
- Pim
- Aris
- Scps
- Qnx
- An
- Ipcomp
- Snp
- Compaqpeer
- Ipxinip
- Vrrp
- Pgm
- Any0Hopprotocol
- L2Tp
- Ddx
- Iatp
- Stp
- Srp
- Uti
- Smp
- Sm
- Ptp
- Isisoveripv4
- Fire
- Crtp
- Crudp
- Sscopmce
- Iplt
- Sps
- Pipe
- Sctp
- Fc
- Rsvpe2Eignore
- Mobilityheader
- Udplite
- Mplsinip
- Manet
- Hip
- Shim6
- Wesp
- Rohc
- Ethernet
- Reserved
